lib/gossip/index.js

```diff
@@ -94,7 +94,6 @@ Gossip.prototype.start = function start() {
         return;
     }
 
-    this.ringpop.membership.shuffle();
```

Comment:
It is good that the ping mechanism and the syncer use different iterators, but as discussed the iterator should slightly change so that it inserts newly added nodes at a random position in the  MembershipIterator.members to conform to chapter 4.3 of the swim paper again.

Before the changes to the lazy shuffle mechanism the line at https://github.com/uber/ringpop-node/blob/syncer/lib/membership/index.js#L285 made sure that newly added nodes have a fair chance of being pinged by already existing nodes, now that the list only gets shuffled when exhausted this fairness went away.

I think it can be solved by making the MembershipIterator listen `updated` events on the membership. But maybe it is easier to add a new member event in this block: https://github.com/uber/ringpop-node/blob/syncer/lib/membership/index.js#L277-L294 since you specifically want to add only new members to the list of the iterator.